java-strings/src/main/java/com/baeldung/string/ReplaceCharacterInString.java

```diff
@@ -0,0 +1,21 @@
+package com.baeldung.string;
+
+public class ReplaceCharacterInString {
+    public String replaceCharSubstring(String str, char ch, int index) {
+        String myString = str.substring(0, index) + ch + str.substring(index);
```

Comment:
```suggestion
        String myString = str.substring(0, index) + ch + str.substring(index + 1);
```

Reply:
I think you need to add 1 to index in the last part, otherwise you're just inserting a character in front of the one you're intending to replace.